test/Common/SupportedLanguagesTestAttribute.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+using System;
+using System.Collections.Generic;
+using Xunit.Sdk;
+using System.Reflection;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Sql.Tests.Common
+{
+    /// <summary>
+    /// SqlInlineData attribute class for adding language parameter data to tests. This allows
+    /// tests to be run against all the supported languages list that are specified in the SupportedLanguages
+    /// list of this class (ex: CSharp, JavaScript)
+    /// </summary>
+    public class SqlInlineDataAttribute : DataAttribute
+    {
+        private readonly List<object[]> testData = new List<object[]>();
+
+        /// <summary>
+        /// Adds a language parameter to the test data which will contain the language 
+        /// that the test is currently running against from the list of supported languages
+        /// </summary>
+        /// <param name="args">The test parameters to insert inline for the test</param>
+        public SqlInlineDataAttribute(params object[] args)
+        {
+            foreach (SupportedLanguages lang in Enum.GetValues(typeof(SupportedLanguages)))
+            {
+                var listOfValues = new List<object>();
+                foreach (object val in args)
+                {
+                    listOfValues.Add(val);
+                }
+                listOfValues.Add(lang);
+                this.testData.Add(listOfValues.ToArray());
```

Comment:
I'm having trouble understanding why we're adding both `lang` and `val` to listOfValues just from reading this. Can you clarify?

Reply:
We have inline data coming in the form of [ [a, b, 3], [d, e, 6]...] and we need to translate that to each language -> 
[ [a, b, 3, CSharp], [a, b, 3, JavaScript], [a, b, 3, Python], ...,
  [d, e, 6, CSharp], [d, e, 6, JavaScript], [d, e, 6, Python], ....
  ...
]
So creating a list of values and adding each supported language. Hope that explains it better. Let me know if you need more clarity.

Reply:
Yeah, I hear ya. Can you add more comments in the code?